lib/grp.gi

```diff
@@ -236,13 +238,15 @@ InstallMethod( IsPGroup,
     # known *and* the group knows to be nilpotent or is abelian;
     # thus an `IsAbelian' test may be forced (which can be done via comparing
     # products of generators) but *not* an `IsNilpotent' test.
-    if     ( not HasSize( G ) )
-       and (    ( HasIsNilpotentGroup( G ) and IsNilpotentGroup( G ) )
-             or IsAbelian( G ) ) then
+    if HasSize( G ) and IsFinite( G ) then
```

Comment:
Why the ````IsFinite```` checks here, and below? If the group knows that it is infinite, then ````IS_PGROUP_FROM_SIZE```` will return false, as it should. This is in correlation with the gap definition of p-group, which should always be finite. However, it contradicts the general definition of p-group, as mathematicians do consider infinite p-groups, as well. (There was some discussion about this one or two years ago, and I think the decision was to keep p-groups finite.)

Reply:
While `IsPGroup` indeed implies `IsFinite` right now, my goal is to change this, as it runs counter against the usual definition of the term in mathematics, which is an obstruction for adding support for infinite p-groups. It turns out that very little actually needs to be changed to pull it off. 

Point is: As long as `IsPGroup` implies `IsFinite`, this part of the change should be harmless, but if we switch the definition, it'll ensure that `IS_PGROUP_FROM_SIZE` isn't called for an infinite group (for which it can't possible say anything in the hypothetical new definition).

Reply:
Very well, agreed.